airbyte-integrations/connectors/source-netsuite/source_netsuite/source.py

```diff
@@ -0,0 +1,84 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, List, Mapping, Tuple
+
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from netsuitesdk import NetSuiteConnection
+from source_netsuite.streams import (
+    Accounts,
+    Classifications,
+    Currencies,
+    Customers,
+    Departments,
+    Employees,
+    ExpenseCategories,
+    ExpenseReports,
+    Files,
+    Folders,
+    JournalEntries,
+    Locations,
+    Projects,
+    Subsidiaries,
+    TaxGroups,
+    TaxItems,
+    Terms,
+    VendorBills,
+    VendorPayments,
+    Vendors,
+)
+
+
+class SourceNetsuiteSoap(AbstractSource):
+    def get_netsuite_connector(self, config: Mapping[str, Any]) -> NetSuiteConnection:
+        return NetSuiteConnection(
+            account=config["realm"],
+            consumer_key=config["consumer_key"],
+            consumer_secret=config["consumer_secret"],
+            token_key=config["token_id"],
+            token_secret=config["token_secret"],
+        )
+
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
```

Comment:
If the `NetSuiteConnection` object verifies the state of the connection (either Connected or Not Connected) it's not required, otherwise - I agree with @davydov-d, this should be done using try-except block along with verification of the actual response (or similar). Please see other connectors to find more examples.

Reply:
If the `NetSuiteConnection` object verifies the state of the connection (either Connected or Not Connected) it's not required, otherwise - I agree with @davydov-d, this should be done using try-except block along with verification of the actual response (or similar). Please see other connectors to find more examples.